src/vm/eventpipe.cpp

```diff
@@ -522,7 +519,7 @@ void EventPipe::WriteEventInternal(EventPipeEvent &event, EventPipeEventPayload
     CONTRACTL_END;
 
     // We can't proceed if tracing is not initialized.
-    if (!s_tracingInitialized)
+    if (s_state == EventPipeState::NotInitialized || s_state == EventPipeState::FailedToInitialize)
```

Comment:
```suggestion
    if (s_state <= EventPipeState::FailedToInitialize)
```

to be consistent with what you have below